mypy/main.py

```diff
@@ -482,6 +528,48 @@ def add_invertible_flag(flag: str,
         print("Warning: --no-fast-parser no longer has any effect.  The fast parser "
               "is now mypy's default and only parser.")
 
+    # Infer Python version and/or executable if one is not given
+    if special_opts.python_executable is not None and special_opts.python_version is not None:
+        try:
+            py_exe_ver = _python_version_from_executable(special_opts.python_executable)
+
+        except PythonExecutableInferenceError as e:
+            parser.error(str(e))
+            sys.exit(2)
+        else:
+            if py_exe_ver != special_opts.python_version:
+                parser.error(
+                    'Error: Python version {} did not match executable {}, got version {}.'.format(
+                        special_opts.python_version, special_opts.python_executable, py_exe_ver
+                    ))
+                sys.exit(2)
+            else:
+                options.python_version = special_opts.python_version
+                options.python_executable = special_opts.python_executable
+    elif special_opts.python_executable is None and special_opts.python_version is not None:
+        options.python_version = special_opts.python_version
+        if not special_opts.no_site_packages:
+            try:
+                py_exe = _python_executable_from_version(special_opts.python_version)
+                options.python_executable = py_exe
+            except PythonExecutableInferenceError as e:
+                if not special_opts.no_site_packages:
```

Comment:
This condition is always true

Reply:
Is it? If the user passes `--no-site-packages`, it should be `not True`, since the flag's action is `store_true`. I could be mistaken in this.

Reply:
Yes, because your code is currently:
```
if x:
    ...
    if x:
        ...
```

I should have said "if this line runs at all, then the condition is true"

Reply:
Oh... yes that would do it.